common/djangoapps/student/views/management.py

```diff
@@ -1014,7 +1018,17 @@ def activate_account(request, key):
             extra_tags='account-activation aa-icon'
         )
     else:
-        if not registration.user.is_active:
+        if not registration.user.is_active and waffle().is_enabled(PREVENT_AUTH_USER_WRITES):
+            messages.error(
+                request,
+                HTML(u'{html_start}{message}{html_end}').format(
+                    message=Text(SYSTEM_MAINTENANCE_MSG),
+                    html_start=HTML('<p class="message-title">'),
+                    html_end=HTML('</p>'),
+                ),
+                extra_tags='account-activation aa-icon',
+            )
+        elif not registration.user.is_active:
```

Comment:
Nit: I'd prefer that the `if not registration.user.is_active` check happened once and the waffle flag check happened in that code block. But - on the fence.

Reply:
I think I'll refactor it to start with the `if registration.user.is_active` case first; the condition is more intuitive that way anyway.